.npmignore

```diff
@@ -1,21 +0,0 @@
-.gitignore
-.jscsrc
-.jshintignore
-.jshintrc
-.travis.yml
-.validate.json
-
-d2.iml
-
-./config.js
-gulpfile.js
-
-.idea
-docs
-jspm_packages/*
-node_modules/*
-coverage/*
-src/*
-test/*
-build/*
-dist/*
```

Comment:
Looks like we should be able to add this file back, I think with the same contents as before (or at least just `./src/**` and `./.eslint*`), and it will be picked up by `npm`?

https://docs.npmjs.com/cli/v6/using-npm/developers#keeping-files-out-of-your-package

Reply:
Yeah, I thought so too, but I cannot get it to work consistently across `yarn pack` and `npm pack`, and I think it comes down to a paragraph in the [NPM package.json docs](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#files):

> You can also provide a `.npmignore` file in the root of your package or in subdirectories, which will keep files from being included. **At the root of your package it will not override the "files" field, but in subdirectories it will.** The `.npmignore` file works just like a `.gitignore`. If there is a `.gitignore` file, and `.npmignore` is missing, `.gitignore`'s contents will be used instead.

So because of these two lines in the `files` array: https://github.com/dhis2/d2/pull/236/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R69-R70

That overrides the rules in the `.npmignore`. I also tried dropping in `.npmignore` into `src`, but I couldn't get it to work consistently. Frankly I am at the point where I've tried so many combinations I am having trouble keeping the results apart so I might be off on some details on previous experiments.

Reply:
Tested again, `.npmignore` doesn't change anything.